implementation/src/main/java/io/smallrye/jwt/KeyUtils.java

```diff
@@ -137,25 +138,55 @@ public static PublicKey decodeJWKSPublicKey(String jwksValue) throws GeneralSecu
      * @throws GeneralSecurityException on decode failure
      */
     public static PublicKey decodePublicKey(String pemEncoded) throws GeneralSecurityException {
-        pemEncoded = removeBeginEnd(pemEncoded);
+        pemEncoded = removeKeyBeginEnd(pemEncoded);
         byte[] encodedBytes = Base64.getDecoder().decode(pemEncoded);
 
         X509EncodedKeySpec spec = new X509EncodedKeySpec(encodedBytes);
         KeyFactory kf = KeyFactory.getInstance(RSA);
         return kf.generatePublic(spec);
     }
 
+    /**
+     * Decode a PEM encoded certificate string to an RSA PublicKey
+     * 
+     * @param pemEncoded - PEM string for certificate
+     * @return PublicKey
+     * @throws GeneralSecurityException on decode failure
+     */
+    public static PublicKey decodeCertificate(String pemEncoded) throws GeneralSecurityException {
+        pemEncoded = removeCertBeginEnd(pemEncoded);
+        byte[] encodedBytes = Base64.getDecoder().decode(pemEncoded);
+        return CertificateFactory.getInstance("X.509")
+                .generateCertificate(new ByteArrayInputStream(encodedBytes)).getPublicKey();
+    }
+
     /**
      * Strip any -----BEGIN*KEY... header and -----END*KEY... footer and newlines
      * 
      * @param pem encoded string with option header/footer
      * @return a single base64 encoded pem string
      */
-    private static String removeBeginEnd(String pem) {
-        pem = pem.replaceAll("-----BEGIN(.*)KEY-----", "");
-        pem = pem.replaceAll("-----END(.*)KEY-----", "");
+    private static String removeKeyBeginEnd(String pem) {
+        pem = pem.replaceAll("-----BEGIN(.*?)KEY-----", "");
+        pem = pem.replaceAll("-----END(.*?)KEY-----", "");
+        pem = pem.replaceAll("\r\n", "");
+        pem = pem.replaceAll("\n", "");
+        pem = pem.replaceAll("\\\\n", "");
+        return pem.trim();
+    }
+
+    /**
+     * Strip any -----BEGIN*CERTIFICATE... header and -----END*CERTIFICATE... footer and newlines
+     * 
+     * @param pem encoded string with option header/footer
+     * @return a single base64 encoded pem string
+     */
+    private static String removeCertBeginEnd(String pem) {
+        pem = pem.replaceAll("-----BEGIN(.*?)CERTIFICATE-----", "");
+        pem = pem.replaceAll("-----END(.*?)CERTIFICATE-----", "");
         pem = pem.replaceAll("\r\n", "");
         pem = pem.replaceAll("\n", "");
+        pem = pem.replaceAll("\\\\n", "");
```

Comment:
What does this do? Can the string ever contain literal `\n`? (Not the newline character, but backslash followed by "n".)

Reply:
Yes, had to add it to deal with [such certs](https://www.googleapis.com/robot/v1/metadata/x509/securetoken@system.gserviceaccount.com), one of them has been copied to the test resources.

Reply:
Ah, very interesting, thanks!